pkg/sync/kubernetes/kubernetes_sync.go

```diff
@@ -72,36 +88,121 @@ func (k *Sync) Notify(ctx context.Context, c chan<- sync.INotify) {
 		return
 	}
 	k.Logger.Infof("Starting kubernetes sync notifier for resource %s", k.ProviderArgs["featureflagconfiguration"])
-	kubeconfig := os.Getenv("KUBECONFIG")
 
-	// Create the client configuration
-	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	clusterConfig, err := k.buildConfiguration()
 	if err != nil {
+		k.Logger.Errorf("Error building configuration: %s", err)
+	}
+	if err := v1alpha1.AddToScheme(scheme.Scheme); err != nil {
 		k.Logger.Panic(err.Error())
 	}
 
-	if k.ProviderArgs["resyncperiod"] != "" {
-		hr, err := time.ParseDuration(k.ProviderArgs["resyncperiod"])
-		if err != nil {
-			k.Logger.Panic(err.Error())
-		}
-		resyncPeriod = hr
+	k.client, err = client.New(clusterConfig, client.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		k.Logger.Fatalln(err)
 	}
 
-	k.client, err = featureflagconfiguration.NewForConfig(config)
+	clusterClient, err := dynamic.NewForConfig(clusterConfig)
 	if err != nil {
-		k.Logger.Panic(err.Error())
+		log.Fatalln(err)
 	}
 
-	if err := ffv1alpha1.AddToScheme(scheme.Scheme); err != nil {
-		k.Logger.Panic(err.Error())
-	}
+	resource := v1alpha1.GroupVersion.WithResource("featureflagconfigurations")
+	factory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(clusterClient,
+		time.Minute, corev1.NamespaceAll, nil)
```

Comment:
What does the `time.Minute` signify here? Could be worth a comment

Reply:
I added it back as a variable and added some  comments